btf/marshal.go

```diff
@@ -379,19 +384,28 @@ func (e *encoder) deflateVarSecinfos(vars []VarSecinfo) []btfVarSecinfo {
 // The function is intended for the use of the ebpf package and may be removed
 // at any point in time.
 func MarshalMapKV(key, value Type) (_ *Handle, keyID, valueID TypeID, _ error) {
+	if key == nil && value == nil {
+		return nil, 0, 0, nil
+	}
+
 	enc := nativeEncoderPool.Get().(*encoder)
 	defer nativeEncoderPool.Put(enc)
 
 	enc.Reset()
 
-	keyID, err := enc.Add(key)
-	if err != nil {
-		return nil, 0, 0, err
+	var err error
+	if key != nil {
```

Comment:
When is it legit to only have a key but not a value, or vice versa? Datasec? Something else?

Reply:
This actually bubbled up while running the selftests. Map `m_queue` in `map_ptr_kern.c` is a `BPF_MAP_TYPE_QUEUE` and only has type info on the value. And `ringbuf_arr` in `test_ringbuf_multi.c` only has the key defined.